### DISCUSSION TITLE
Dynamic seoMeta

### DISCUSSION THREAD

**Discussion by gregoirepuget** [nuxt/nuxt#18479] [Questions]
Hello,

I would like to do something like this : 
```
const {data: page } = await useFetch(`${apiUrl}/page/${slug}/`)
useHead({
	title: page.head.title
})
definePageMeta({
  layout: page.layout,
});
```
Don't know how to make it work or what i'm doing wrong...
Thanks for your help.

**Comment by TheAlexLichter** [comment]
`page` is a ref. Create a `computed`, e.g. `const title = computed(() => page.value.head.title)` and use it as `useHead({ title })`.

**Comment by ndragun92** [comment]
For me, even better approach would be

```html
<template>
   <Html>
      <Head>
         <Title>{{ page.head.title }}</Title>
      </Head>
   </Html>
</template>

```

It is way cleaner and easier to maintain if logic is simple inside its own component/template/page

**Reply by TheAlexLichter** [reply]
I would advise against using `page.head` directly because it will throw an error if `page` is empty, e.g. because the fetch request failed.

Whether using `useHead` or Meta Components depends solely on preference 😋

**Reply by ndragun92** [reply]
If i have cases like that I usually use `page?.head?.title`
And fallback if need. Optional template params are such an amazing thing in Vue/Nuxt 3 😊
